generated/nidaqmx/_lib.py

```diff
@@ -156,43 +153,48 @@ def cal_handle(self) -> type:
     
     @property
     def encoding(self):
-        if lib_importer.windll.library == 'nicai_utf8' and self._cdll.library == 'nicai_utf8':
-            return 'utf-8'
-        else:
-            return locale.getlocale()[1]
+        if self._encoding is None:
+            self._import_lib()
+        return self._encoding
 
     def _import_lib(self):
         """
         Determines the location of and loads the NI-DAQmx CAI DLL.
         """
         self._windll = None
         self._cdll = None
+        self.encoding = None
```

Comment:
`self.encoding` is a read-only property. This looks like a typo for `self._encoding`.

[functools.cached_property](https://docs.python.org/3/library/functools.html#functools.cached_property) has a tricky way of "overwriting" a property with a simple value but I think it uses `__dict__` to do the "overwrite".